routing: treat SERIAL/LOCAL_SERIAL consistency as LWT for routing (#1647)
scylladb/scylla-rust-driver: docs/source/load-balancing/default-policy.md

[^1]: There is an optimisation implemented for LWT requests that routes them
to the replicas in the ring order (as it prevents contention due to Paxos conflicts), so replicas in that case are not shuffled in groups at all.

> The optimisation is applied in two cases: (1) when an LWT statement (containing an `IF` clause) is prepared — ScyllaDB sets a flag that the driver uses for routing; (2) when a statement is executed with `Serial` or `LocalSerial` consistency — this is the only way to execute `SELECT` as LWT (since it has no `IF` clause), and the driver detects it automatically.

Reviewer: ```suggestion
The optimisation is applied in two cases:
1. when an LWT statement (containing an `IF` clause) is prepared — ScyllaDB sets a flag that the driver uses for routing;
2. when a statement is executed with `Serial` or `LocalSerial` consistency (this is the only way to execute `SELECT` as LWT, since it may not contain an `IF` clause), the driver detects it automatically.
```
